pkg/oauth/oauthclient_test.go

```diff
@@ -0,0 +1,395 @@
+		},
+		{
+			expectedError:       fmt.Errorf("error: OAuthClient object %s not found", oauthClientName),
+			oauthTestClientName: oauthClientName,
+			addToRuntimeObjects: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testOAuthClient := generateOAuthClient(testCase.oauthTestClientName)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testOAuthClient)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		// Test the Pull function
+		builderResult, err := Pull(testSettings, testCase.oauthTestClientName)
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestOAuthClientUpdate(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testOAuthClient := generateOAuthClient(oauthClientName)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testOAuthClient)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		oauthClientBuilder, err := Pull(testSettings, oauthClientName)
+		assert.Nil(t, err)
+
+		// Create a change in the builder
+		annotation := map[string]string{"test": "test"}
+		oauthClientBuilder.Definition.Annotations = annotation
+
+		// Test the Update function
+		builderResult, err := oauthClientBuilder.Update()
+
+		assert.Equal(t, err, testCase.expectedError)
+
+		// Validate that the resource was updated
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+			assert.Equal(t, builderResult.Object.Annotations, annotation)
+		}
+	}
+}
+
+func TestOAuthClientCreate(t *testing.T) {
+	testCases := []struct {
+		addToRuntimeObjects bool
+	}{
+		{
+			addToRuntimeObjects: true,
+		},
+		{
+			addToRuntimeObjects: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, &oauthv1.OAuthClient{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: oauthClientName,
+				},
+			})
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		testBuilder, client := generateOAuthClientBuilder(testSettings, oauthClientName)
+
+		// Test the Delete function
+		_, err := testBuilder.Create()
+		assert.Nil(t, err)
+
+		// Assert that the object actually exists
+		_, err = Pull(client, oauthClientName)
```

Comment:
Instead I think you should just get register the builder value returned from the Create method and just assert that builder.Object is not Nil